packages/cli/src/server/api/memory/groups.ts

```diff
@@ -0,0 +1,134 @@
+        const roomName = name || `Chat ${new Date().toLocaleString()}`;
+
+        await runtime.ensureWorldExists({
+          id: worldId,
+          name: source,
+          agentId: runtime.agentId,
+          serverId: serverId as UUID,
+        });
+
+        await runtime.ensureRoomExists({
+          id: roomId,
+          name: roomName,
+          source,
+          type: ChannelType.API,
+          worldId,
+          serverId: serverId as UUID,
+          metadata,
+          channelId: roomId,
+        });
+
+        await runtime.addParticipant(runtime.agentId, roomId);
+        await runtime.ensureParticipantInRoom(runtime.agentId, roomId);
+        await runtime.setParticipantUserState(roomId, runtime.agentId, 'FOLLOWED');
+
+        results.push({
+          id: roomId,
+          name: roomName,
+          createdAt: Date.now(),
+          source: 'client',
+          worldId,
+        });
+      } catch (error) {
+        logger.error(`[ROOM CREATE] Error creating room for agent ${agentId}:`, error);
+        errors.push({
+          agentId,
+          code: error.message === 'Agent not found' ? 'NOT_FOUND' : 'CREATE_ERROR',
+          message: error.message === 'Agent not found' ? error.message : 'Failed to Create group',
+          details: error.message,
+        });
+      }
+    }
+
+    if (results.length === 0 && errors.length > 0) {
+      res.status(500).json({
+        success: false,
+        error: errors.length
+          ? errors
+          : [{ code: 'UNKNOWN_ERROR', message: 'No rooms were created' }],
+      });
+      return;
+    }
+
+    res.status(errors.length ? 207 : 201).json({
+      success: errors.length === 0,
+      data: results,
+      errors: errors.length ? errors : undefined,
+    });
+  });
+
+  // Delete group
+  router.delete('/groups/:serverId', async (req, res) => {
+    const worldId = validateUuid(req.params.serverId);
+    if (!worldId) {
+      return sendError(res, 400, 'INVALID_ID', 'Invalid serverId (worldId) format');
+    }
+    if (!db) {
+      return sendError(res, 500, 'DB_ERROR', 'Database not available');
+    }
+
+    try {
+      await db.deleteRoomsByWorldId(worldId);
+      res.status(204).send();
+    } catch (error) {
+      logger.error('[GROUP DELETE] Error deleting group:', error);
+      sendError(res, 500, 'DELETE_ERROR', 'Error deleting group', error.message);
+    }
+  });
+
+  // Clear group memories
+  router.delete('/groups/:serverId/memories', async (req, res) => {
+    const worldId = validateUuid(req.params.serverId);
+    if (!worldId) {
+      return sendError(res, 400, 'INVALID_ID', 'Invalid serverId (worldId) format');
+    }
+    if (!db) {
+      return sendError(res, 500, 'DB_ERROR', 'Database not available');
+    }
+
+    try {
+      const memories = await db.getMemoriesByWorldId({ worldId, tableName: 'messages' });
+      for (const memory of memories) {
+        await db.deleteMemory(memory.id as UUID);
+      }
```

Comment:
We should write a `deleteManyMemories` batch query function, would be much faster in SQL